lib/moesif_rack/moesif_middleware.rb

```diff
@@ -36,6 +36,60 @@ def initialize app, options = {}
       @config_dict = Hash.new
       @disable_transaction_id = options['disable_transaction_id'] || false
       @log_body = options.fetch('log_body', true)
+      @batch_size = options['batch_size'] || 25
+      @events_queue = Queue.new
+      @event_response_config_etag = nil
+
+      Thread::new do
+        loop do
+          begin
+            if @events_queue.size > 0
+              has_events = true
+            else
+              has_events = false
+            end
+  
+            until has_events == false do 
```

Comment:
may be if  u change code to until events_queue.empty, then you dont need if checks inside and also code above it, no need of variable has_events?